lib/dao.js

```diff
@@ -52,6 +52,44 @@ function getIdValue(m, data) {
   return data && data[idName(m)];
 }
 
+function copyData(from, to) {
+  for (var key in from) {
+    to[key] = from[key];
+  }
+}
+
+function convertDataPropertiesByType(data) {
+  var typedData = {};
+  for (var key in data) {
+    // Convert the properties by type
+    typedData[key] = data[key];
+    if (typeof typedData[key] === 'object'
+            && typedData[key] !== null
+            && typeof typedData[key].toObject === 'function') {
+      typedData[key] = typedData[key].toObject();
+    }
+  }
+  return typedData;
+}
+
+function strictCheck(model, strict, data, inst, cb) {
+  var props = model.definition.properties;
+  var keys = Object.keys(data);
+  var result = {}, key;
+  for (var i = 0; i < keys.length; i++) {
+    key = keys[i];
+    if (props[key]) {
+      result[key] = data[key];
+    } else if (strict === 'throw') {
+      cb(new Error('Unknown property: ' + key));
+      return;
+    } else if (strict === 'validate') {
+      inst.__unknownProperties.push(key);
```

Comment:
Yeah, it's not great. Could you please add a comment before `strictCheck`, explain what the method is doing and add a notice that it will modify `inst` when `strict` is `validate`?